server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java

```diff
@@ -993,6 +994,40 @@ public void testSupportsParallelCollection() {
             searchSourceBuilder.aggregation(new TermsAggregationBuilder("terms"));
             assertFalse(searchSourceBuilder.supportsParallelCollection(fieldCardinality));
         }
+        {
+            SearchSourceBuilder searchSourceBuilder = newSearchSourceBuilder.get();
+            searchSourceBuilder.aggregation(new TopHitsAggregationBuilder("tophits"));
+            assertTrue(searchSourceBuilder.supportsParallelCollection(fieldCardinality));
+        }
+        {
+            SearchSourceBuilder searchSourceBuilder = newSearchSourceBuilder.get();
+            searchSourceBuilder.aggregation(new TopHitsAggregationBuilder("tophits").sort("_score"));
+            assertTrue(searchSourceBuilder.supportsParallelCollection(fieldCardinality));
+        }
+        {
+            SearchSourceBuilder searchSourceBuilder = newSearchSourceBuilder.get();
+            searchSourceBuilder.aggregation(new TopHitsAggregationBuilder("tophits").sort(SortBuilders.fieldSort("field")));
+            assertFalse(searchSourceBuilder.supportsParallelCollection(fieldCardinality));
+        }
+        {
+            SearchSourceBuilder searchSourceBuilder = newSearchSourceBuilder.get();
+            searchSourceBuilder.aggregation(new TermsAggregationBuilder("terms").subAggregation(new TopHitsAggregationBuilder("tophits")));
+            assertFalse(searchSourceBuilder.supportsParallelCollection(fieldCardinality));
```

Comment:
Is this test unrelated to the "sort" change made here? Do we generally disallow parallel collection for nested aggs or is there something specific here?

Reply:
yea I wanted to test the different combinations of top_hits within another agg as well as other agg with top_hits underneath.